Add accessibility labels to FormToggle

justinshreve opened the pull request:
Our `<FormToggle>` component does not currently allow you to set labels for the on/off state. This PR adds two new props to the component to allow you to do that when using the full (not compact) toggle button. Also includes 2 tests for FormToggle.

Having these labels can make the toggles more accessible to those who might have a difficult time telling the state just based on colors or position. (I sometimes personally have trouble immediately recognizing the state of these kinds of buttons and have the iOS accessibility feature for this turned on for example). These labels can also help provide context, in general, for what the toggle will do (Enabled/Disabled, On/Off, etc).

When going through our mockups for WooCommerce, I noticed this was not currently possible and thought it would be a good exercise for one of my first contributions to Calypso.

The CSS might also need a little help cc @kellychoffman.

--

The buttons do not change if you do not include the label props, so this should not affect any existing buttons.

To Test:
* View the example at https://calypso.live/devdocs/design/form-fields?branch=add/formtoggle-labels

drw158 commented on client/components/forms/form-toggle/style.scss:
```diff
@@ -69,6 +73,21 @@
 	}
 
+	& + .form-toggle__label .form-toggle__switch .form-toggle__label-text {
+		float: right;
```
Instead of using these floats, we should be able to use `flexbox`.

drw158 commented:
Interesting, looks good. Make sure that the Off text color is hight enough contrast against the gray background. I'd use white text on the blue background for On.

I would argue that there is not a reason to add these labels only for certain toggles and not others. The flexibility of the prop is not worth the inconsistency. We should probably do this for all non-compact toggles. I'd like to hear some more opinions though.

MichaelArestad commented:
I agree with @drw158. Whatever we do, it should be applied to all labels. That said, I don't think this change is super necessary. I could see it as a prop that is enabled via an "accessibility mode" of sorts. Alternatively, if you want to improve accessibility, you could make these labels screen reader only unless a user is in accessibility mode.

MichaelArestad commented:
I'm definitely all about us figuring out an accessibility mode, by the way. We could up the color contrast, make things a bit bigger maybe, and, of course, make these meta labels visible.

kellychoffman commented:
Toggles without labels are a standard pattern at this point.

Adding the labels as screen reader text on all toggles sounds like a good step to take until we have an accessibility mode people can turn on. (Similar to how iOS adds `1` and `0` to their toggles.)

justinshreve commented:
Updated the color and added an example to the forms section of devdocs for a quick demo.

Doing something like this as part of an accessibility mode instead makes sense. It would definitely be useful to be able to enable them for all buttons as you said. 

I'm happy to update this to a simpler PR that just sets screen reader text for the buttons in the meantime. It was a fun experiment as I'm ramping up on Calypso stuff :).

MichaelArestad commented:
@davidakennedy Can you verify that adding these as screenreader text would be helpful and not annoying for screenreaders?
